test/context/onerror.js

```diff
@@ -169,5 +171,23 @@ describe('ctx.onerror(err)', () => {
         .expect('Content-Type', 'text/plain; charset=utf-8')
         .expect('Internal Server Error', done);
     });
+
+    it('should use res.getHeaderNames() accessor when available', () => {
```

Comment:
@fengmk2 something like this? I avoided using context helper to get the real `http.ServerResponse`